contracts/StakingBank.sol

```diff
@@ -3,61 +3,59 @@ pragma solidity ^0.6.8;
 
 import '@openzeppelin/contracts/cryptography/MerkleProof.sol';
 import '@openzeppelin/contracts/utils/ReentrancyGuard.sol';
+import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
 import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
 
-import "./ValidatorRegistry.sol";
+import "./interfaces/IERC20Detailed.sol";
+import "./interfaces/IStakingBank.sol";
+import "./interfaces/IValidatorRegistry.sol";
 
-contract StakingBank is ReentrancyGuard {
+contract StakingBank is IStakingBank, ERC20, ReentrancyGuard {
   using SafeERC20 for IERC20;
+  using SafeMath for uint256;
 
-  address public tokenAddress;
-  address public registryAddress;
+  IERC20 public token;
+  IValidatorRegistry public registry;
 
   constructor(
     address _tokenAddress,
     address _registryAddress
-  ) public {
-    tokenAddress = _tokenAddress;
-    registryAddress = _registryAddress;
+  ) ERC20(
+      string(abi.encodePacked("staked ", IERC20Detailed(_tokenAddress).name())),
```

Comment:
This is pretty strange to me. Is this following a standard or something?

Reply:
I would not call it standard yet, but I saw this solution in DeFi contracts and I liked it. Basically you need a name for token, and when you put ie. `UMB` on stake how would you name this staked token so it would be descriptive? `staked Umbrella` and `sbUMB` (sb stands for StakingBank) sound logical to me. 

And the code itself for achieving it is basically forced by solidity, because you operating with different types of strings, thats why this `abi` stuff.

Reply:
I see ok, let's make the name a bit prettier. How about: "Staked Umbrella" and "SB-UMB". Do you need to add space? Otherwise would it be: "stakeUmbrella" and "sbUMB"?

Reply:
Actually let's do: "Staked Umbrella" for the name and "SUMB".